cumulus_library_catalog/catalog/icd10_annotated_counts.py

```diff
@@ -1,61 +1,93 @@
-            table_cols=[
-                "code",
-            ],
+            table_name=table_name,
             source_table="core__condition",
+            table_cols=["code", "system"],
             where_clauses=[
                 # These codes are excluded from the catalog because they correspond
-                # to sensitive conditions (mental health/STDs) that hospitals 
+                # to sensitive conditions (mental health/STDs) that hospitals
                 # do not want to share with others for legal reasons
-                "block_code not in('A50-A64','A70-A74','B20-B20')",
-                "chapter_code not in('F01-F99','Z00-Z99')",
-                f"cnt_subject_ref >= {min_subject}"
-                ],
-            annotation= cumulus_library.CountAnnotation(
-                    field = 'code',
-                    join_table= '"umls"."icd10_hierarchy"',
-                    join_field = 'leaf_code',
-                    columns= ([    
-                        ('chapter_code', None),
-                        ('chapter_str', None),
-                        ('block_code', None),
-                        ('block_str', None),
-                        ('category_code', None),
-                        ('category_str', None),
-                        ('subcategory_1_code', None),
-                        ('subcategory_1_str', None),
-                        ('subcategory_2_code', None),
-                        ('subcategory_2_str', None),
-                        ('subcategory_3_code', None),
-                        ('subcategory_3_str', None),
-                        ('extension_code', None),
-                        ('extension_str', None),
-                    ])
-                ),
-            )
+                "block_code NOT IN ('A50-A64','A70-A74','B20-B20')",
+                "chapter_code NOT IN ('F01-F99','Z00-Z99')",
+                f"cnt_subject_ref >= {min_subject}",
+                f"{target_col} IS NOT NULL",
+                "system ='http://hl7.org/fhir/sid/icd-10-cm'",
+            ],
+            annotation=cumulus_library.CountAnnotation(
+                field="code",
+                join_table='"umls"."icd10_hierarchy"',
+                join_field="leaf_code",
+                columns=annotation_cols,
+                alt_target=target_col,
+            ),
+        )
+
     def prepare_queries(
-        self, 
-        *args, 
+        self,
+        *args,
         manifest: study_manifest.StudyManifest | None = None,
-        min_subject: int | None=None,
+        min_subject: int | None = None,
         **kwargs,
     ):
         self.queries.append(
-            self.get_icd10_diagnosis_query(min_subject)
+            "CREATE TABLE catalog__count_icd10_all AS ("
+            "SELECT count(DISTINCT subject_ref) AS cnt FROM core__condition "
+            "WHERE system ='http://hl7.org/fhir/sid/icd-10-cm');"
         )
+        for table_info in (
```

Comment:
Could maybe simplify this with `for depth, table_name in enumerate(["chapter", "block", ...])`

(And either pass in `depth + 1` or make the depth zero-counted in the called method.)